src/common/FrameManager.ts

```diff
@@ -73,7 +73,7 @@ export class FrameManager extends EventEmitter {
   private _frames = new Map<string, Frame>();
   private _contextIdToContext = new Map<string, ExecutionContext>();
   private _isolatedWorlds = new Set<string>();
-  private _mainFrame: Frame;
+  private _mainFrame!: Frame;
```

Comment:
where is _mainFrame set? it looks like whether we have it is dependent on CDP. Let's keep it optional here and assert it when we need to use it?